File: sdk/go-guest/templates/greeting.go
// lsf-example-begin: capsule
package export_examples_greeting_api

import (
    "strings"
    wit "go.bytecodealliance.org/pkg/wit/types"
)

func Greet(name string) wit.Result[string, string] {
    name = strings.TrimSpace(name)
    if len(name) == 0 { return wit.Err[string, string]("Please enter a name.") }
    if len(name) > 100 { return wit.Err[string, string]("Use a name of at most 100 bytes.") }
    return wit.Ok[string, string]("Hello, " + name + "!")
}
// lsf-example-end: capsule

File: sdk/java-guest/templates/greeting.java
package dev.latent.app;
import dev.latent.generated.Bindings;
import dev.latent.guest.Result;
import dev.latent.guest.Text;

// lsf-example-begin: capsule
public final class Capsule implements Bindings.Exports {
    public Result<String, String> greet(String name) {
        String trimmed = Text.trim(name);
        if (trimmed.isEmpty()) return Result.err("Please enter a name.");
        if (Text.utf8Length(trimmed) > 100) return Result.err("Use a name of at most 100 bytes.");
        return Result.ok("Hello, " + trimmed + "!");
    }
}
// lsf-example-end: capsule
